### PR TITLE
Added zoom features:

### PR DESCRIPTION
**Pull request by Jacoblairm**
- Reset camera zoom, reset value can be configured.
- Camera zoom speed.
- Preview https://giant.gfycat.com/SlowLoneCrab.webm

### CODE REVIEW

**Comment by Jacoblairm** [comment]
Fixes #919

**Comment by Jacoblairm** [comment]
Fixes #919

File: runelite-client/src/main/java/net/runelite/client/plugins/zoom/ZoomPlugin.java [hunk 1]
```diff
@@ -163,7 +173,17 @@ public void keyReleased(KeyEvent e)
 	{
 		if (e.getKeyCode() == KeyEvent.VK_CONTROL)
 		{
-			controlDown = false;
+			switch (zoomConfig.controlFunction())
+			{
+				case CONTROL_TO_ZOOM:
+					controlDown = false;
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/zoom/ZoomPlugin.java]: This should probably be outside of the switch still

File: runelite-client/src/main/java/net/runelite/client/plugins/zoom/ZoomPlugin.java [hunk 2]
```diff
@@ -77,7 +82,7 @@ public void onScriptEvent(ScriptCallbackEvent event)
 		int[] intStack = client.getIntStack();
 		int intStackSize = client.getIntStackSize();
 
-		if ("scrollWheelZoom".equals(event.getEventName()) && zoomConfig.requireControlDown() && !controlDown)
+		if ("scrollWheelZoom".equals(event.getEventName()) && zoomConfig.controlFunction().equals(ControlFunction.CONTROL_TO_ZOOM) && !controlDown)
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/zoom/ZoomPlugin.java]: You can use `==` for enum comparisons

**Comment by Nightfirecat** [comment]
My one complaint is that this offers no option to require ctrl to scroll and have some kind of option to scroll back to a default level. I'm not sure how that would work exactly, but I'd be interested in some way to utilize both features.

**Comment by jplsek** [comment]
Could make it so that ctrl+middle click would reset the zoom. Just need to be careful that moving the camera with middle click doesn't trigger resetting the zoom.

File: runelite-client/src/main/java/net/runelite/client/plugins/zoom/ZoomPlugin.java [hunk 3]
```diff
@@ -173,16 +174,11 @@ public void keyReleased(KeyEvent e)
...
 			{
-				case CONTROL_TO_ZOOM:
-					controlDown = false;
-					break;
-				case CONTROL_TO_RESET:
-					int zoomValue = zoomConfig.ctrlZoomValue();
-					clientThread.invokeLater(() -> client.runScript(ScriptID.ZOOM_INPUT, zoomValue, zoomValue));
-					controlDown = false;
-					break;
+				int zoomValue = zoomConfig.ctrlZoomValue() > INNER_ZOOM_LIMIT ? INNER_ZOOM_LIMIT : zoomConfig.ctrlZoomValue();
```
[the remark this answers is not in the document]
Comment (Jacoblairm) [reply runelite-client/src/main/java/net/runelite/client/plugins/zoom/ZoomPlugin.java]: Didn't think you could set anything less than 0 in the settings?

[the remark this answers is not in the document]
Comment (Abextm) [reply runelite-client/src/main/java/net/runelite/client/plugins/zoom/ZoomPlugin.java]: I know you can set zero, and that is the value that I am mostly worried about.

File: runelite-client/src/main/java/net/runelite/client/plugins/zoom/ZoomPlugin.java [hunk 4]
```diff
@@ -164,6 +176,23 @@ public void keyReleased(KeyEvent e)
 		if (e.getKeyCode() == KeyEvent.VK_CONTROL)
 		{
 			controlDown = false;
+
+			if (zoomConfig.controlFunction() == ControlFunction.CONTROL_TO_RESET)
+			{
+				int finalZoomValue;
+				int zoomValue = zoomConfig.ctrlZoomValue();
```
Comment (Abextm) [inline runelite-client/src/main/java/net/runelite/client/plugins/zoom/ZoomPlugin.java]: Please just do a normal bounds check.

Comment (Jacoblairm) [reply runelite-client/src/main/java/net/runelite/client/plugins/zoom/ZoomPlugin.java]: So you're saying I shouldn't clamp the value and allow the zoom reset to occur if it's out of bounds?

Comment (Abextm) [reply runelite-client/src/main/java/net/runelite/client/plugins/zoom/ZoomPlugin.java]: `zoomValue` is mutable. `finalZoomValue` has no purpose and should not exist.

**Pull request closed without merging**